main.go

```diff
@@ -1,18 +1,186 @@
+		return
+	}
+}
+
+// SELECT all guitars from database
+func dbQueryAllGuitars() []Guitar {
+	db := dbConnection()
+	var multipleGuitars []Guitar
+	// Query all Guitars from db
+	sql := "SELECT * FROM guitars "
+	rows, err := db.Query(sql)
+	if err != nil {
+		fmt.Printf("Error Query, and %s", err)
+	}
+
+	for rows.Next() {
+		var eachGuitar Guitar
+		err = rows.Scan(&eachGuitar.Id, &eachGuitar.Name, &eachGuitar.Brand_id, &eachGuitar.Year, &eachGuitar.Description)
+		if err != nil {
+			fmt.Printf("error Looping data, and %s", err)
+		}
+		multipleGuitars = append(multipleGuitars, eachGuitar)
+	}
+	return multipleGuitars
+}
+
+// GET request to return data from dbReturnAllGuitars()
+func getAllGuitars(w http.ResponseWriter, r *http.Request) {
+	data := dbQueryAllGuitars()
+	w.Header().Set("Content-type", "application/json")
+	fmt.Println("Get single guitar endpoint hit")
+	err := json.NewEncoder(w).Encode(data)
+	if err != nil {
+		http.Error(w, err.Error(), 500)
+		return
+	}
+}
+
+// ? POST request INSERTING a guitar to the database
+func addGuitar(w http.ResponseWriter, r *http.Request) {
+	db := dbConnection()
+	defer db.Close()
+	w.Header().Set("Content-type", "application/json")
+
+	var guitar Guitar
+	err := json.NewDecoder(r.Body).Decode(&guitar)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	sqlStatement := `
+	INSERT INTO guitars (name, brand_id, year, description)
+	VALUES ($1, $2, $3, $4)
+	returning id`
+	id := 0
+	err = db.QueryRow(sqlStatement, guitar.Name, guitar.Brand_id, guitar.Year, guitar.Description).Scan(&id)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	w.WriteHeader(http.StatusCreated)
+	fmt.Fprintf(w, "Item with ID %d was created", id)
+}
+
+// DELETE request
+func deleteGuitar(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-type", "application/json")
+	db := dbConnection()
+
+	// To retrieve a particular record form the database, we need to pass an id paremeter to the URL. We will use the following methods and assign it to the urlId variable
+	urlId := r.URL.Query().Get("id")
+
+	// To add a layer of security, we will cast the urlId param to an integer from a string. This will be passed into the database query below.
+	urlIdInt, err := strconv.Atoi(urlId)
+	if err != nil {
+		panic(err)
+	}
+
+	sqlStatement := "DELETE FROM guitars WHERE id = $1;"
+
+	res, err := db.Exec(sqlStatement, urlIdInt)
+	if err != nil {
+		panic(err)
+	}
+	count, err := res.RowsAffected()
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println(count)
+	defer db.Close()
+}
+
+func homePage(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintf(w, "GuitarAPI Project Home Page")
+	fmt.Println("Endpoint Hit: Home Page")
 }
 
 func handleRequests() {
 	http.HandleFunc("/", homePage)
+	http.HandleFunc("/guitar", getSingleGuitar)
+	http.HandleFunc("/guitars", getAllGuitars)
+	http.HandleFunc("/new", addGuitar)
```

Comment:
This URL `/new` is not RESTful, so this will need to change ([hint / explanation why](https://learn.microsoft.com/en-us/azure/architecture/best-practices/api-design#organize-the-api-design-around-resources))